File: stringy/surodenci.py
from sys import stdin, stdout

vstup = list(stdin.readline())
del vstup[-1]
pocet = [int(i) for i in stdin.readline().split(' ')][0]

#skontroluj ci je symetricky
je_symetricke = True

polka = len(vstup) // 2
for letter in range(polka):
    if vstup[letter] != vstup[-letter-1]:
        je_symetricke = False
        break

res = []

for _ in range(pocet):
    pozicia, pismeno = [i for i in input().split(' ')]
    pozicia = int(pozicia)
    vstup[pozicia] = pismeno

    if vstup[pozicia] == vstup[-pozicia-1] and je_symetricke:
        res.append('ano')
    else:
        for letter in range(polka):
            if vstup[letter] != vstup[-letter-1]:
                je_symetricke = False
                res.append('nie')
                break
        else:       
            res.append('ano')
            je_symetricke = True

stdout.write("\n".join(res))
stdout.write("\n")

File: stringy/surodenci.cpp
#include <iostream>
#include <string>
using namespace std;

int main()
{
	string vstup;
	cin >> vstup;
	int n = vstup.size();
	int poln = n / 2;
	
	int pocet;
	cin >> pocet;
	
	bool je_symetricke = true;
	
	for (int i = 0; i < poln; i++)
	{
		if (vstup[i] != vstup[n-1-i])
		{
			je_symetricke = false;
			break;
		}
	}
	
	int index;
	char znak;
	
	while (pocet--)
	{
		cin >> index;
		cin >> znak;
		vstup[index] = znak;
		
		if (vstup[index] == vstup[n-1-index])
		{
			cout << "ano" << endl;
		}
		else
		{
			bool ok = true;
			for (int i = 0; i < poln; i++)
			{
				if (vstup[i] != vstup[n-1-i])
				{
					je_symetricke = false;
					cout << "nie" << endl;
					ok = false;
					break;
				}
			}
			if (ok == true)
			{
				cout << "ano" << endl;
				je_symetricke = true;	
			}	
		}
		
	}	
}
